### PR TITLE
Revamp README.md to be single-page.

### PR DESCRIPTION
**Pull request by JonDouglas**
Refactors the `README.md` to be single-page and provide links for ways to directly contribute and provide feedback.

These items have been discussed as part of our community engagement efforts. 
 
https://github.com/xamarin/xamarin-macios has already implemented these changes as of last Friday.

### CODE REVIEW

**Comment by jonpryor** [comment]
Do not like.

Here's where I'm coming from. Assume Joe Random Developer does:

```
git clone https://github.com/xamarin/xamarin-android.git
cd xamarin-android
make prepare
```

Then disappears on a plane flight or otherwise has no/limited network connectivity.

I want the resulting repo to be at least *minimally* useful for providing information on how to build, test, and debug, *without* requiring that the developer go on line to view that information.

This is why we have e.g. [Documentation/DevelopmentTips.md](https://github.com/xamarin/xamarin-android/blob/master/Documentation/DevelopmentTips.md) *within the repo*, and not on some wiki page. Ditto [Documentation/build_process.md](https://github.com/xamarin/xamarin-android/blob/master/Documentation/build_process.md) (though that's equally so that when e.g. a new Build action is added, the PR adding the build action can also minimally document it).

*This* PR removes ***all** useful information related to building the product*, all to get a "shorter" README.md.

I am not enamored with this idea.

I'm not opposed to a shorter README, but I want all the build information to say *within the repo*.

The Wiki should be only for information that is not tied to repo "state" or commit. Rephrased: a future commit may change how the product is built. By having the build documentation *within* the repo, changing this is straightforward, and has the added bonus that when the build system is updated, the associated documentation can be updated as part of the same commit.

If the build documentation is in a Wiki, the Wiki instead needs to know the history of every part of the build system. If things change such that we can e.g. "just" load `Xamarin.Android.sln` into Visual Studio and hit **Build** in d15-8, the Wiki will need to mention that this is *only* true for d15-8, and in prior releases something else must be done, etc.

**Comment by JonDouglas** [comment]
## Goals

My goal here is to clearly outline how a user can **contribute** and **provide feedback** to Xamarin.Android.

My hope by making these changes is to help evolve the OSS repo's documentation by formally documenting tasks surrounding **contribution** and **feedback** in a browseable wiki. In this case, using GitHub's wiki feature.

## Background

Here's my rationale that probably should've been in the original PR description as this is a **"who moved my cheese"** type of PR:

Now that GitHub Issues receives majority of the repo traffic, people are stumbling upon a large `README.md` when hitting the repo home page. 

Here's a few examples of goals that a user might have when coming to the repository:

**1. User wants to contribute to the repository.**

The `README.md` had items listed in terms of the process for building and running from source, but does not let the user know how they can effectively contribute such as:

- ~How to build and run from source~
- The project/development workflow Xamarin.Android uses
- Debugging (User would have to discover "DevelopmentTips.md")
- Testing (User would have to discover "DevelopmentTips.md")
- ~Coding guidelines~
- How to submit a Pull Request
- How to get started with a `good-first-issue` or other smaller tasks as first contributions

**2. User wants to provide feedback for the repository.**

Currently, the `README.md` lists [two ways to report bugs](https://github.com/xamarin/xamarin-android#reporting-bugs). From my background, we can do much better and let the user know exactly where they can go to get us the feedback we need to improve or discuss the product. This helps point the user in the correct direction depending on their feedback and prevents issues from being filed if they better fit in a different place(For example a question that might fit better on Stack Overflow or the Forums). I started [documenting this effort in the wiki](https://github.com/xamarin/xamarin-android/wiki/Submitting-Bugs,-Feature-Requests,-and-Pull-Requests), and after researching other repositories, it made sense to surface at the `README.md` level as well.

## Summary

This PR attempts to make the `README.md` concise with an "at a glance" table of contents representing each task a user might be faced when **contributing** or **providing feedback**. It serves as a foundation to formally document the repository and encourage contribution.

For example, **everything that was removed from the `README.md`** was refactored to a [How to Contribute](https://github.com/xamarin/xamarin-android/wiki/How-to-Contribute) wiki that helps the user understand how they can contribute via:

- Building and running Xamarin.Android
- Using the Xamarin.Android Build
- Samples
- Development Tips

I hope that this [How to Contribute](https://github.com/xamarin/xamarin-android/wiki/How-to-Contribute) wiki article will evolve as the project evolves and includes more helpful items for contributors such as:

- Debugging (How one might debug certain parts of the repo like MSBuild tasks/targets, binding generator, etc)
- Testing (How one might create and run unit tests around their PRs)
- Development Workflow (Best practices for keeping the repository clean)
- Areas of contribution (Curated issues that others can actively pickup)
- etc

If this is too significant of a change, I can re-work these items after more discussion.

**Note on offline support**

Wikis can also be cloned by appending `.wiki` when cloning a repo.

**EX:**

```
git clone https://github.com/xamarin/xamarin-android.wiki.git
```

**TLDR; Trying to refactor `README.md` so it's more concise with links to wiki areas based on the user's goal to contribute or provide feedback when visiting the repository.**

**Comment by jonpryor** [comment]
> My goal here is to clearly outline how a user can contribute and provide feedback to Xamarin.Android.

Which is a wonderful goal. And if a shorter README helps there, yay.

That does *not* in turn imply that the information should be *removed from the repo and **moved** into the wiki*.

Additionally, being able to clone the wiki does not, in my mind, actually help, because it requires that the developer actually think to do so.

For example, Linux has build documentation *within the source tree* (along with lots of other stuff): https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/Documentation

CoreCLR contains build documentation *within the source tree*: https://github.com/dotnet/coreclr/tree/master/Documentation

(Not the Wiki. The source tree.)

CoreFX contains build documentation *within the source tree*: https://github.com/dotnet/corefx/tree/master/Documentation/building

Roslyn -- you guessed it! -- contains documentation *within the source tree*: https://github.com/dotnet/roslyn/blob/master/docs/contributing/Building,%20Debugging,%20and%20Testing%20on%20Windows.md

Do you quite understand what I'm getting at?

I'm fine with a shorter README.

I want *all the documentation* which is removed from the README to remain *within this repository*. **Not** the wiki.

**Comment by jonpryor** [comment]
It should also be noted, in case it isn't clear, that [GitHub Markdown supports relative links](https://blog.github.com/2013-01-31-relative-links-in-markup-files/).

This is *already done* within the current `README.md`, e.g.

```markdown
Windows users will need to use the `setup-windows.exe` tool as described in
[`Documentation/UsingJenkinsBuildArtifacts.md`](Documentation/UsingJenkinsBuildArtifacts.md#oss-xamarinandroidzip-installation).
```

very nicely links to [`Documentation/UsingJenkinsBuildArtifacts.md`](Documentation/UsingJenkinsBuildArtifacts.md).

**Comment by JonDouglas** [comment]
I completely understand where you're coming from and wanted to throw this idea out in the wild to discuss on. I'll rework these changes eventually to reflect your thoughts on where documentation should best be included.

It brings up the question:

**Does it make sense to use a wiki in any fashion for this repo?**

**Some don't:**

https://github.com/dotnet/coreclr/wiki
https://github.com/dotnet/corefx/wiki

**Others do:**

https://github.com/dotnet/roslyn/wiki
https://github.com/Microsoft/TypeScript/wiki

The difference tends to be related around items that support the product but have little or no impact on the repository once downloaded. For example, you might see **repo policies, roadmap, how to submit bugs in a wiki.**

Whereas you might see **build requirements, debugging, testing, etc inside a Documentation folder**.

Thoughts?

**Comment by jonpryor** [comment]
> Does it make sense to use a wiki in any fashion for this repo?

Yes, and I outlined the criteria [previously](https://github.com/xamarin/xamarin-android/pull/1343#issuecomment-369018388)

> The Wiki should be only for information that is not tied to repo "state" or commit

For example:

* <https://github.com/xamarin/xamarin-android/wiki/APK-Tests-on-the-Hyper-V-Emulator>

    This is "exploratory" and "explanatory" text on running `.apk` test within a Hyper-V emulator. It *could* be within the repo, but it is not truly tied to any "state" within the repo. I'm fine with this being in the wiki.

* <https://github.com/xamarin/xamarin-android/wiki/Design-Time-Build-System>

    Information about Visual Studio's Design-Time Build System, which tends to screw things up 6 ways from Sunday every time we interact with it. *Much* of that information is *also* in commit messages, e.g. 3b2cce27769fed9f987f8f65f09db02718dbf41b, but this document is also "explanatory" for *what Visual Studio does*, and is thus a "living" document, not tied to repo state.

* <https://github.com/xamarin/xamarin-android/wiki/Test-Cloud>

    Information on running tests within Xamarin Test Cloud. As we are not currently executing any tests within Xamarin Test Cloud, it is likewise not tied to repo state. This document is *explanatory* in nature.

All of the above I'm fine with bing in the Wiki, and I explicitly asked that at least two of them be placed in the wiki.

**Comment by jonpryor** [comment]
As an example, consider PR #1353, which adds support for a new `$(API_LEVEL)` make variable for building. Previous commits won't be able to use this variable, so if the build documentation which mentioned `$(API_LEVEL)` lived in the Wiki, we'd need to mention in which commit it was added in, lest someone try to use it in e.g. d15-7 and find that it had no effect.

Build system changes + relevant documentation should be in the same commit. Having build system documentation in the Wiki makes this impossible.

**Comment by jonpryor** [comment]
Superseded by PR #1358.
